src/account/default.ts

```diff
@@ -573,23 +574,59 @@ export class Account extends Provider implements AccountInterface {
     return getMessageHash(typedData, this.address);
   }
 
-  public async verifyMessageHash(hash: BigNumberish, signature: Signature): Promise<boolean> {
+  public async verifyMessageHash(
+    hash: BigNumberish,
+    signature: Signature
+  ): Promise<SignatureVerifResult> {
     try {
-      await this.callContract({
+      const resp = await this.callContract({
         contractAddress: this.address,
         entrypoint: 'isValidSignature',
         calldata: CallData.compile({
           hash: toBigInt(hash).toString(),
           signature: formatSignature(signature),
         }),
       });
-      return true;
-    } catch {
-      return false;
+      // console.log('verifySign=', resp);
+      if (BigInt(resp.result[0]) === 0n) {
+        // OpenZeppelin 0.8.0 invalid signature
+        return {
+          isVerificationProcessed: true,
+          isSignatureValid: false,
+        } as SignatureVerifResult;
+      }
+      // OpenZeppelin 0.8.0, ArgentX 0.3.0 & Braavos Cairo 0 valid signature
+      return {
+        isVerificationProcessed: true,
+        isSignatureValid: true,
+      } as SignatureVerifResult;
+    } catch (err) {
+      // console.log('verifySign error=', err);
+      if ((err as Error).message.includes('argent/invalid-signature')) {
+        // ArgentX 0.3.0 invalid signature
+        return {
+          isVerificationProcessed: true,
+          isSignatureValid: false,
+        } as SignatureVerifResult;
+      }
+      if ((err as Error).message.includes('is invalid, with respect to the public key')) {
+        // Braavos Cairo 0 invalid signature
+        return {
+          isVerificationProcessed: true,
+          isSignatureValid: false,
+        } as SignatureVerifResult;
+      }
+      return {
+        isVerificationProcessed: false,
+        error: new Error('Signature verification request is rejected by the network.'),
+      } as SignatureVerifResult;
```

Comment:
```suggestion
      if (BigInt(resp.result[0]) === 0n) {
        // OpenZeppelin 0.8.0 invalid signature
        return false;
      }
      // OpenZeppelin 0.8.0, ArgentX 0.3.0 & Braavos Cairo 0 valid signature
      return true;
    } catch (err) {
      if (['argent/invalid-signature','is invalid, with respect to the public key'].includes((err as Error).message)) {
        // ArgentX 0.3.0 invalid signature, Braavos Cairo 0 invalid signature
        return false;
      }
      throw Error(`Signature verification request is rejected by the network: ${err}`)
```

Reply:
Also, this seems like a future issue for maintaining, are there any standards for this (if not we should propose one)? We can't cover all wallets and devs' Acc implementations in lib.

Reply:
SNIP-6 is existing : https://github.com/starknet-io/SNIPs/blob/main/SNIPS/snip-6.md
I don't know if this SNIP is accepted, but nobody is conform to it, neither for the function name, not for the return type.
So, I covered the 3 main account classes (ArgentX, Braavos, Openzeppelin).